apps/mula/lib/components.ex

```diff
@@ -78,4 +78,78 @@ defmodule Mula.Components do
     </div>
     """
   end
+
+  attr(:value, :float,
+    required: true,
+    doc:
+      "the value represented on the meter, must be between min_value and max_value, if min_value and max_value are not passed, value will be a plain percentage"
+  )
+
+  attr(:max_value, :float,
+    default: 100.0,
+    doc:
+      "the maximun value of the interval represented on the meter, must be greater than min_value, must be greater than or equal value"
+  )
+
+  attr(:min_value, :float,
+    default: 0.0,
+    doc:
+      "the minimun value of the interval represented on the meter, must be lesser than max_value, must be lesser than or equal value"
+  )
+
+  attr(:value_text_formatter, :any,
+    default: nil,
+    doc:
+      "a function that receives the calculated percentage and returns a string, it's used to set the aria-valuetext and the value_text returned"
+  )
+
+  slot(:inner_block, doc: "the optional inner block that renders the")
+
+  def meter(assigns) do
+    %{
+      value: value,
+      max_value: max_value,
+      min_value: min_value,
+      value_text_formatter: value_text_formatter
+    } = assigns
+
+    unless max_value > min_value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] max_value must be greater than min_value"
+    end
+
+    unless max_value >= value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] max_value must be greater than or equal to value"
+    end
+
+    unless value >= min_value do
+      raise ArgumentError,
+        message: "[Mula.Components.meter/1] min_value must be lesser than or equal to value"
+    end
+
+    percentage = (value - min_value) / (max_value - min_value) * 100
+
+    value_text =
+      unless is_nil(value_text_formatter),
+        do: apply(value_text_formatter, [percentage]),
+        else: "#{percentage}%"
+
+    assigns =
+      assign(assigns, %{
+        value: value,
+        min_value: min_value,
+        max_value: max_value,
+        percentage: percentage,
+        value_text: value_text
+      })
+
+    ~H"""
+    <div role="meter" aria-valuenow={@value} aria-valuemin={@min_value} aria-valuemax={@max_value} aria-valuetext={@value_text}>
+      <%= render_slot(@inner_block, %{min_value: @min_value, max_value: @max_value, percentage: @percentage, value_text: @value_text}) %>
```

Comment:
Here I think only percentage and value_text should be given to the caller since the rest of the assigns the caller already has.